benchmark/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+add_subdirectory(jsonschema)
```

Comment:
Now that I realise, you are activating this file if `JSONTOOLKIT_BENCHMARK` is set, but then for each of these subdirectories, you need to check the relevant component variables. For example, here, you should only enable this directory if `JSONTOOLKIT_JSONSCHEMA`, etc? 

That means we also need to gracefully handle what happens if you enable benchmarks but no actual components. I guess you can just exit with a `FATAL_ERROR` but you could also create a custom target that prints a useful string like "no benchmarks detected" and that does an exit 1?

Reply:
> you should only enable this directory if JSONTOOLKIT_JSONSCHEMA, etc?

that should be enough no? I mean regarding this:

> That means we also need to gracefully handle what happens if you enable benchmarks but no actual components. I guess you can just exit with a FATAL_ERROR but you could also create a custom target that prints a useful string like "no benchmarks detected" and that does an exit 1?

Reply:
I let a todo. I think we could explore that later and focus on tackling the performance issue first.